src/include/buffer.h

```diff
@@ -318,6 +323,15 @@ namespace buffer CEPH_BUFFER_API {
       void copy(unsigned len, list &dest);
       void copy(unsigned len, std::string &dest);
       void copy_all(list &dest);
+
+      friend bool operator==(const iterator_impl& lhs,
+			     const iterator_impl& rhs) {
+	return lhs.bl == rhs.bl && lhs.off == rhs.off;
```

Comment:
I see, sorry I didn't look closely enough to see the visibility("hidden") yesterday.

This looks fine then. Just need to be sure that the potentially inlined iterator::operators will still work if we change them or iterator_impl in the future. I doubt that will be necessary though.